stl/inc/format

```diff
@@ -1248,15 +1243,449 @@ _OutputIt _Write(_OutputIt _Out, basic_string_view<_CharT> _Value) {
+    _STL_ASSERT(_Ec == errc{}, "to_chars failed");
+
+    auto _Buffer_start = _Buffer.data();
+    auto _Width        = static_cast<int>(_End - _Buffer_start);
+
+    if (_Value >= _Integral{0}) {
+        if (_Specs._Sgn != _Sign::_Minus) {
+            _Width += 1;
+        }
+    } else {
+        // Remove the '-', it will be dealt with directly
+        _Buffer_start += 1;
+    }
+
+    if (_To_upper) {
+        _Fmt_buffer_to_uppercase(_Buffer_start, _End);
+    }
+
+    string_view _Prefix;
+    if (_Specs._Alt) {
+        _Prefix = _Get_integral_prefix(_Specs._Type, _Value);
+        _Width += static_cast<int>(_Prefix.size());
+    }
+
+    const bool _Write_leading_zeroes = _Specs._Leading_zero && _Specs._Alignment == _Align::_None;
+    auto _Writer                     = [&, _End = _End](_OutputIt _Out) {
+        _Out = _Write_sign(_Out, _Specs._Sgn, _Value < _Integral{0});
+        _Out = _STD copy(_Prefix.begin(), _Prefix.end(), _Out);
+        if (_Write_leading_zeroes && _Width < _Specs._Width) {
+            _Out = _STD fill_n(_Out, _Specs._Width - _Width, '0');
+        }
+        return _STD copy(_Buffer_start, _End, _Out);
+    };
+
+    if (_Write_leading_zeroes) {
+        return _Writer(_Out);
+    }
+
+    return _Write_aligned(_Out, _Width, _Specs, _Align::_Right, _Writer);
+}
+#pragma warning(pop)
+
+// clang-format off
+template <class _CharT, class _OutputIt, integral _Integral>
+    requires (!_CharT_or_bool<_Integral, _CharT>)
+_OutputIt _Write(_OutputIt _Out, _Integral _Value, _Basic_format_specs<_CharT>& _Specs) {
+    // clang-format on
+    return _Write_integral(_Out, _Value, _Specs);
+}
+
+template <class _CharT, class _OutputIt>
+_OutputIt _Write(_OutputIt _Out, bool _Value, _Basic_format_specs<_CharT>& _Specs) {
+    if (_Specs._Type && _Specs._Type != 's') {
+        // Standard says to cast to unsigned char, but this is just going to be wider and prevent an extra template
+        // instantiation.
+        return _Write_integral(_Out, static_cast<unsigned int>(_Value), _Specs);
+    }
+
+    if (_Specs._Precision != -1) {
+        throw format_error("bool cannot have a precision");
+    }
+
+    if constexpr (is_same_v<_CharT, wchar_t>) {
+        return _Write(_Out, _Value ? L"true" : L"false", _Specs);
+    } else {
+        return _Write(_Out, _Value ? "true" : "false", _Specs);
+    }
+}
+
+template <class _CharT, class _OutputIt>
+_OutputIt _Write(_OutputIt _Out, _CharT _Value, _Basic_format_specs<_CharT>& _Specs) {
+    if (_Specs._Type && _Specs._Type != 'c') {
+        return _Write_integral(_Out, _Value, _Specs);
+    }
+
+    if (_Specs._Precision != -1) {
+        throw format_error("charT cannot have a precision");
+    }
+
+    // Clear the type so that the string_view writer doesn't fail on 'c'.
+    _Specs._Type = 0;
+    return _Write(_Out, basic_string_view<_CharT>(&_Value, 1), _Specs);

```

Comment:
Style: Recommend braces to construct temporaries (we're less consistent for multi-arg but I still think it's a good idea).